shared/actions/authActions.js

```diff
@@ -153,71 +188,182 @@ export const setToken = (token) => {
-    return await AsyncStorage.getItem(STORAGE_KEY_refreshToken);
-  } catch (error) {
-    console.log(error.message);
+export const setProfileImgUrl = (profileImgUrl) => {
+  return {
+    type: types.setProfileImgUrl,
+    profileImgUrl
   }
 };
+// todo : pass it to grantfbuser after receiving 400
+export const requestRefreshTokenFacebook = async (refreshToken) => {
+  const address = 'https://goober.herokuapp.com/api/auth/refresh';
+  const bodyRequestRefreshToken = JSON.stringify({
+    'refreshToken': refreshToken
+  });
+  fetch(address, {
+    method: 'POST',
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    },
+    body: {
+      'refreshToken': bodyRequestRefreshToken
+    }
+  })
+  .then((response) => {
+    if (response.status === 200) {
+      return response.json();
+    } else if (response.status === 400) {
+      removeUserToken();
+    }
+  })
+  .then((rjson) => {
+    console.log(rjson);
+    setAccessToken(rjson.accessToken);
+    setRefreshToken(rjson.refreshToken);
+  })
+  .catch((error) => {
+    console.log(error);
+  })
+};
 
-export const setUserToken = async (response, userType) => {
-  try {
-    await AsyncStorage.setItem(STORAGE_KEY_accessToken, response.accessToken);
-    await AsyncStorage.setItem(STORAGE_KEY_refreshToken, response.refreshToken);
-    if (userType === 'anonymous') {
-      await AsyncStorage.setItem(STORAGE_KEY_secret, response.secret);
+export const requestRefreshTokenGuest = async (refreshToken) => {
+  const address = 'https://goober.herokuapp.com/api/auth/refresh';
+  const bodyRequestRefreshToken = JSON.stringify({
+    'refreshToken': refreshToken
+  });
+  fetch(address, {
+    method: 'POST',
+    headers: {
+      'Accept': 'application/json',
+      'Content-Type': 'application/json'
+    },
+    body: bodyRequestRefreshToken
+  })
+  .then((response) => {
+    if (response.status === 200) {
+      return response.json();
+    } else if (response.status === 400) {
+      getSecretToken().then((secret) => {
+        if (secret !== null) {
+          getUserKey().then((userId) => {
+            if (userId !== null) {
+              grantAnonymousUser(secret, userId);
+            }
+          });
+        } else {
+          signupGuestUser();
+        }
+      });
     }
-  } catch (error) {
-    console.log(error.message);
-  }
+  })
+  .then((rjson) => {
+    console.log(rjson);
+    setAccessToken(rjson.accessToken);
+    setRefreshToken(rjson.refreshToken);
+  })
+  .catch((error) => {
+    console.log(error);
+  })
 };
 
-export const removeUserToken = async () => {
+export const getUserInformation = async (userKey, accessToken) => {
+  console.log(userKey);
+  console.log(accessToken);
+  const address = `http://goober.herokuapp.com/api/users/${userKey}`;
+  const headerGetUserInformation = JSON.stringify({
+    'Accept': 'application/json',
+    'Content-Type': 'application/json',
+    'authorization': `bearer ${accessToken}`
```

Comment:
this header is used quite many times, so why don't you just make it as a constant or a function.